src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,187 @@
+            for ground_nsrt in utils.all_ground_nsrts(nsrt, objects)
+        ]
+
+        def policy(
+            sete: FrozenSet[GroundAtom]
+        ) -> Optional[Dict[FrozenSet[GroundAtom], _GroundNSRT]]:
+            if dictionary and sete in dictionary:
+                return dictionary[sete]
+            return None
+
+        def check_goal(atoms: FrozenSet[GroundAtom]) -> bool:
+            return goal.issubset(atoms)
+
+        def get_valid_actions(
+            atoms: FrozenSet[GroundAtom]
+        ) -> Iterator[Tuple[_GroundNSRT, float]]:
+            for op in utils.get_applicable_operators(ground_nsrts, atoms):
+                yield (op, 1.0)
+
+        def get_next_state(atoms: FrozenSet[GroundAtom],
+                           ground_nsrt: _GroundNSRT) -> FrozenSet[GroundAtom]:
+            return frozenset(utils.apply_operator(ground_nsrt, set(atoms)))
+
+        heuristic = utils.create_task_planning_heuristic(
+            heuristic_name=CFG.pg3_task_planning_heuristic,
+            init_atoms=init,
+            goal=goal,
+            ground_ops=ground_nsrts,
+            predicates=self._initial_predicates,
+            objects=set(state),
+        )
+        _, action_seq = utils.run_policy_guided_astar(
+            initial_state=initial_atoms,
+            check_goal=check_goal,
+            get_valid_actions=get_valid_actions,
+            get_next_state=get_next_state,
+            heuristic=heuristic,
+            policy=policy,
+            num_rollout_steps=CFG.pg3_max_policy_guided_rollout,
+            rollout_step_cost=0)
+
+        return action_seq
+
+    def _get_llm_based_state_action_predictions(
+            self, atoms: Set[GroundAtom], goal: Set[GroundAtom],
+            state: State) -> List[Tuple[List[Any], List[Any]]]:
+        trajectories = []
+        initial_atoms = atoms
+
+        new_prompt = self._create_prompt(atoms, goal, [])
+        prompt = self._prompt_prefix + new_prompt
+        llm_predictions = self._llm.sample_completions(
+            prompt=prompt,
+            temperature=CFG.open_loop_llm_temperature,
+            seed=CFG.seed,
+            num_completions=CFG.open_loop_llm_num_completions)
+
+        for llm_prediction in llm_predictions:
+            objects = set(state)
+            option_plan = self._llm_prediction_to_option_plan(
+                llm_prediction, objects)
+            if len(option_plan) == 0:
+                continue
+            states, actions = self._get_nsrt_ground_plan(
+                initial_atoms, option_plan)
+
+            trajectories.append((states, actions))
+        return trajectories
+
+    def _get_nsrt_ground_plan(
+        self, initial_atoms: Set[GroundAtom],
+        option_plan: List[Tuple[ParameterizedOption, Sequence]]
+    ) -> Tuple[List, List]:
+
+        states = []
+        actions = []
+        states.append(initial_atoms)
+        nsrts = self._get_current_nsrts()
+        #We assume that operators and NSRTs are the same
```

Comment:
please match the format of our other comments: spaces in between hashtags and the words, and periods at the end of the sentence